### DISCUSSION TITLE
Can tree components be supported？

### DISCUSSION THREAD

**Discussion by Ali-ovo** [TanStack/table#4525] [Ideas]
I can feel the lightness of this component, but because of work reasons, the tree component must be used. Can you support this function？ Thank you !

Tree compnents be like this : https://ant.design/components/table-cn/#components-table-demo-tree-data，
Similar to this component：  https://tanstack.com/table/v8/docs/examples/react/expanding，
but the open level can be deeper

If the function has been implemented, can you tell me how to implement it? Thank you.

**Comment by KevinVandy** [comment]
You can use any UI you want with TanStack Table. It's just up to you to be able to hook up the logic with the UI components you want

**Reply by Ali-ovo** [reply]
Do I need to render recursively on the columns value of the component? I think this is an extremely dangerous thing.
But if I don't do this, I can't render his children node. Obviously, it is not a tree component
